### PR TITLE
Cache auth secret and allow configuring parallel uploads

### PR DESCRIPTION
**Pull request by divolgin**

### CODE REVIEW

File: kustomize/overlays/dev/deployment.yaml [hunk 1]
```diff
@@ -71,6 +71,8 @@ spec:
               value: /etc/kurl-proxy/ca/tls.crt
             - name: KOTS_INSTALL_ID
               value: dev-1pu4oeY162e2pbLpK4JubK6hxrX
+            - name: AIRAGAP_UPLOAD_PARALLELISM
```
Comment (salahalsaleh) [inline kustomize/overlays/dev/deployment.yaml:74]: spelling :)
```
AIRGAP_UPLOAD_PARALLELISM
```

**Review by salahalsaleh — APPROVED** [review]

File: web/src/components/UploadAirgapBundle.jsx [hunk 2]
```diff
@@ -51,13 +52,38 @@ class UploadAirgapBundle extends React.Component {
...
+}
+
   uploadAirgapBundle = async () => {
     const { match, showRegistry } = this.props;
 
     // Reset the airgap upload state
     const resetUrl = `${window.env.API_ENDPOINT}/app/${match.params.slug}/airgap/reset`;
     try {
-      await fetch(resetUrl, {
+      let res = await fetch(resetUrl, {
```
Comment (salahalsaleh) [inline web/src/components/UploadAirgapBundle.jsx:86]: why?

File: web/src/components/apps/AppVersionHistory.jsx [hunk 3]
```diff
@@ -75,13 +75,13 @@ class AppVersionHistory extends Component {
   }
 
   componentWillMount() {
```
Comment (salahalsaleh) [inline web/src/components/apps/AppVersionHistory.jsx:77]: remove?

File: web/src/components/apps/AppVersionHistory.jsx [hunk 4]
```diff
@@ -75,13 +75,13 @@ class AppVersionHistory extends Component {
...
-    const { app } = this.props;
-    if (app.isAirgap) {
-      this.airgapUploader = new AirgapUploader(true, app.slug, this.onDropBundle);
-    }
   }
 
   componentDidMount() {
+    if (this.props.app.isAirgap && !this.state.airgapUploader) {
```
Comment (salahalsaleh) [inline web/src/components/apps/AppVersionHistory.jsx:81]:
```suggestion
    if (this.props.app?.isAirgap && !this.state.airgapUploader) {
```

**Review by salahalsaleh — APPROVED** [review]

**Review by salahalsaleh — APPROVED** [review]

File: cmd/kots/cli/install.go [hunk 5]
```diff
@@ -393,6 +397,9 @@ func InstallCmd() *cobra.Command {
 	cmd.Flags().Bool("ensure-rbac", true, "when set, kots will create the roles and rolebindings necessary to manage applications")
 	cmd.Flags().MarkHidden("ensure-rbac")
 
+	cmd.Flags().String("airgap-upload-parallelism", "", "the number of chunks to upload in parallel when installing or updating in airgap mode")
```
Comment (emosbaugh) [inline cmd/kots/cli/install.go:400]: why string and not int?

**Pull request merged**